pkg/querier/worker/scheduler_processor.go

```diff
@@ -363,6 +371,11 @@ sendBody:
 	for offset := 0; offset < len(response.Body); {
 		select {
 		case <-reqCtx.Done():
+			if cause := context.Cause(reqCtx); cause != nil {
+				level.Warn(logger).Log("msg", "response stream aborted", "cause", cause)
+			} else {
+				level.Warn(logger).Log("msg", "response stream aborted", "err", reqCtx.Err())
```

Comment:
I don't think we need the `if` here. I think context.Cause will use reqCtx.Err() if there's no explicit cause

```
// Cause returns a non-nil error explaining why c was canceled.
// The first cancellation of c or one of its parents sets the cause.
// If that cancellation happened via a call to CancelCauseFunc(err),
// then [Cause] returns err.
// Otherwise Cause(c) returns the same value as c.Err().
// Cause returns nil if c has not been canceled yet.
func Cause(c Context) error {
```